src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -380,32 +380,45 @@ export class WebRTCAdaptor
 		}
 	}
 
-	publish(streamId, token) 
+	publish(streamId, token, subscriberId, subscriberCode) 
 	{
-		//If it started with playOnly mode and wants to publish now
-		if(this.localStream == null){
-			this.navigatorUserMedia(this.mediaConstraints, (stream => {
-				this.gotStream(stream);
-				var jsCmd = {
+		var jsCmd = null;
+		if(subscriberId !== undefined && subscriberCode !== undefined) {
+			jsCmd = {
```

Comment:
Just have single json object and use subscriberId and subscriberCode if it exists like this 
```
subscriberId: typeof subscriberId !== undefined ? subscriberId : "" ,
subscriberCode: typeof subscriberCode !== undefined ? subscriberCode : "",
```